how-to/common/public/views/app/app-definition-builder/app-definition-builder-view.js

```diff
@@ -0,0 +1,214 @@
+		name: id.value,
+		title: title.value,
+		description: description.value,
+		manifestType: selectedManifestType.manifestType
+	};
+
+	if (selectedManifestType.manifest === 'string') {
+		app.manifest = manifest.value;
+	} else if (selectedManifestType.manifest === 'object') {
+		try {
+			if (inlineManifest.value.trim() !== '') {
+				app.manifest = JSON.parse(inlineManifest.value);
+			} else {
+				app.manifest = {};
+			}
+		} catch {
+			// dynamically typing json. Do not update the setting until it is valid
+			app.manifest = 'Invalid Inline Manifest. Please make it JSON compliant.';
+		}
+	}
+
+	if (icon.value.startsWith('http')) {
+		app.icons = [{ src: icon.value }];
+	} else {
+		app.icons = [];
+	}
+
+	if (appImage.value.startsWith('http')) {
+		app.images = [{ src: appImage.value }];
+	} else {
+		app.images = [];
+	}
+
+	app.contactEmail = contactEmail.value;
+	app.supportEmail = supportEmail.value;
+	app.publisher = publisher.value;
+	app.tags = tags.value.trim() === '' ? [] : tags.value.split(',').map((tag) => tag.trim());
+
+	const selectedIntents = [...availableIntents.selectedOptions].map((option) => option.value);
+	const mappedIntents = [];
+
+	for (let i = 0; i < selectedIntents.length; i++) {
+		const selectedIntent = intents.find((entry) => entry.name === selectedIntents[i]);
+		if (selectedIntent !== undefined) {
+			mappedIntents.push(selectedIntent);
+		}
+	}
+
+	app.intents = mappedIntents;
+
+	return app;
+}
+
+async function updateAppPreview() {
+	const app = await getAppDefinition();
+
+	preview.textContent = JSON.stringify(app, null, 4);
+}
+
+function bindManifestTypes() {
+	const manifestTypeOptions = manifestTypes
+		.map((data, index) => `<option value=${data.manifestType}>${data.displayName}</option>`)
+		.join('\n');
+	manifestType.innerHTML = manifestTypeOptions;
+	bindManifestTypeDetails();
+	updateAppPreview();
+}
+
+function bindManifestTypeDetails() {
+	const manifestTypeInfo = getManifestType();
+	manifestTypeDescription.textContent = manifestTypeInfo.description;
+	if (manifestTypeInfo.helpLink !== undefined && manifestTypeInfo.helpLink !== '') {
+		helpLink.href = manifestTypeInfo.helpLink;
+		helpLink.style.display = 'unset';
+	} else {
+		helpLink.style.display = 'none';
+	}
+	if (manifestTypeInfo.manifest === 'string') {
+		manifestContainer.style.display = 'flex';
+		inlineManifestContainer.style.display = 'none';
+	} else if (manifestTypeInfo.manifest === 'object') {
+		manifestContainer.style.display = 'none';
+		inlineManifestContainer.style.display = 'flex';
+	}
+}
+
+function bindIntents() {
+	const intentsOptions = intents
+		.map((data, index) => `<option value=${data.name}>${data.displayName}</option>`)
+		.join('\n');
+	availableIntents.innerHTML = intentsOptions;
+	updateAppPreview();
+}
+
+function reset() {
+	id.value = '';
+	title.value = '';
+	description.value = '';
+	bindManifestTypes();
+	inlineManifest.value = `{ "url": "https://mydomain.com/", "fdc3InteropApi": "1.2" }`;
```

Comment:
Make this a JSON.stringify of the real object

Reply:
done